source/common/tcp_proxy/upstream.cc

```diff
@@ -96,6 +141,11 @@ void HttpUpstream::encodeData(Buffer::Instance& data, bool end_stream) {
     return;
   }
   request_encoder_->encodeData(data, end_stream);
+
+  if (type_ == Http::CodecType::HTTP1) {
```

Comment:
consider adding a comment to clarify why H1 doesn't continue for below

Reply:
added comment